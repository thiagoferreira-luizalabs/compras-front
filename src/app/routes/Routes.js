import React, { Component } from "react"
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import AppRoute from "./AppRoute"
import App from "../App"
import AuthService from "../services/AuthService"
// Pages
import Login from "../containers/login/Login"
import Theme from "../containers/theme/Theme"
import Home from "../containers/home/Home"

class Routes extends Component {

  constructor(props) {
    super(props)
    this.state = { auth: AuthService.getAuth() }
    this.setAuth = this.setAuth.bind(this)
    this.logout = this.logout.bind(this)
  }

  setAuth() {
    AuthService.setAuth()
    this.setState({ auth: AuthService.getAuth() })
  }

  logout() {
    AuthService.logout()
    this.setState({ auth: AuthService.getAuth() })
    return <Redirect to={{ pathname: '/' }} />
  }

  render() {
    return (
      <Router>
        <Switch>
          <App auth={this.state.auth}>
            {/* Public Routes */}
            <AppRoute public exact {...this.state} path="/" component={Login}
              componentProps={{ setAuth: this.setAuth }} />

            {/* Private Routes */}
            <AppRoute private exact {...this.state} path="/home" component={Home} />
            <AppRoute private exact {...this.state} path="/theme" component={Theme} />

            <AppRoute exact {...this.state} path="/logout" component={() => this.logout()} />

          </App>
        </Switch>
      </Router>
    )
  }

}

export default Routes