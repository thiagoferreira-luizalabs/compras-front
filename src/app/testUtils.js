import React from "react"
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

const MemoryRouterHOC = (children) => (<MemoryRouter>{children}</MemoryRouter>)

const toJsonSnapshot = (wrapper) => toJson(wrapper, { mode: "deep", noKey: true })

export {
  shallow, mount, sinon, MemoryRouter, toJsonSnapshot, asyncAssert, createFileChangeEvent, MemoryRouterHOC
}
