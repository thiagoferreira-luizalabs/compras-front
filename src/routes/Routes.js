import React, { Component } from "react"
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import AppRoute from "./AppRoute"
import App from "../App"
import AuthService from "../services/AuthService"
// Pages
import Login from "../containers/login/Login"
const Home = () => (<h1>Welcome to the homepage. You are logged in! Go corintia. </h1>)

class Routes extends Component {

  constructor(props) {
    super(props)
    this.state = { auth: AuthService.getAuth() }
    this.setAuth = this.setAuth.bind(this)
    this.logout = this.logout.bind(this)
  }

  setAuth() {
    AuthService.setAuth()
    this.setState({ auth: AuthService.getAuth() })
  }

  logout() {
    AuthService.logout()
    this.setState({ auth: AuthService.getAuth() })
    return <Redirect to={{ pathname: '/' }} />
  }

  render() {
    return (
      <Router>
        <Switch>
          <App auth={this.state.auth}>
            {/* Public Routes */}
            <AppRoute public exact {...this.state} path="/" component={Login}
              componentProps={{ setAuth: this.setAuth }} />

            {/* Private Routes */}
            <AppRoute private exact {...this.state} path="/home" component={Home} />

            <AppRoute exact {...this.state} path="/logout" component={() => this.logout()} />

          </App>
        </Switch>
      </Router>
    )
  }

}

export default Routes