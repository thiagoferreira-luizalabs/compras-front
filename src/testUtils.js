import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
// import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

export {
  shallow, mount, sinon,  /*MemoryRouter,*/ toJson, asyncAssert, createFileChangeEvent
}
