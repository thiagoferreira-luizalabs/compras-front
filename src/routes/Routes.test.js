import React from 'react';
import Routes from './Routes';
import { mount, toJsonSnapshot, MemoryRouterHOC, sinon } from "../testUtils"

global.window.localStorage = {}
const localStorage = global.window.localStorage

describe("Routes tests", () => {
  beforeEach(() => {
    localStorage.getItem = sinon.mock()
    localStorage.setItem = sinon.mock()
  })

  it("Render routes for non authenticated user", () => {

    localStorage.getItem.returns(null) // not authenticated

    let wrapper = mount(MemoryRouterHOC(<Routes />))
    expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
    expect(localStorage.getItem.calledOnce).toBeTruthy()
  })


  it("Render routes for authenticated user", () => {
    let auth = btoa(JSON.stringify({
      isAuthenticated: true,
      user: "TBU_FERREIRA",
      token: "XPTO_TOKEN"
    }))

    localStorage.getItem.returns(auth)

    let wrapper = mount(MemoryRouterHOC(<Routes />))
    expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
    expect(localStorage.getItem.calledOnce).toBeTruthy()
  })
})

