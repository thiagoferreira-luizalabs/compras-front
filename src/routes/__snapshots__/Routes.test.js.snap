// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes tests Render routes for authenticated user 1`] = `
<div
  className="App"
>
  <div
    className="ui menu"
  >
    <div
      className="header item"
      onClick={[Function]}
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        Compras
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/theme"
        onClick={[Function]}
      >
        Theme
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/logout"
        onClick={[Function]}
      >
        <button
          className="ui button"
          onClick={[Function]}
          role="button"
        >
          Logout
        </button>
      </a>
    </div>
  </div>
  <h1>
    Welcome to the homepage. You are logged in! Go corintia. 
  </h1>
  <p>
    Version 
    0.1.0
  </p>
</div>
`;

exports[`Routes tests Render routes for non authenticated user 1`] = `
<div
  className="App"
>
  <p>
    Welcome. Insert your credentials to log in.
    <button
      onClick={[Function]}
    >
      Click here to log in
    </button>
  </p>
  <p>
    Version 
    0.1.0
  </p>
</div>
`;
